packages/colorpickers/src/elements/ColorPicker/ColorWell.tsx

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { useRef, useState, useContext } from 'react';
+import { ThemeContext } from 'styled-components';
+import throttle from 'lodash.throttle';
+import { IHSVColor } from './reducer';
+import { hsl2hsv } from '../../utils/conversion';
+import { calculateNextHsv, getSaturationPosition } from '../../utils/saturation';
+import {
+  StyledColorWell,
+  StyledColorWellGradient,
+  StyledColorWellThumb,
+  StyledColorWellWrapper
+} from '../../styled';
+
+interface IColorWellProps {
+  hue: number;
+  saturation: number;
+  lightness: number;
+  onChange?: (hsv: IHSVColor, event: React.MouseEvent<any>) => void;
+}
+
+export const ColorWell: React.FC<IColorWellProps> = ({ hue, saturation, lightness, onChange }) => {
+  const { rtl } = useContext(ThemeContext);
+  const container = useRef<HTMLDivElement>(null);
+  const hsv = hsl2hsv(hue, saturation, lightness);
+  const [position, setPosition] = useState<{ s: number; v: number }>({
+    s: rtl ? 100 - hsv.s : hsv.s,
+    v: hsv.v
+  });
```

Comment:
[nit] not sure if this is complex enough to warrant a [lazy initial state syntax](https://reactjs.org/docs/hooks-reference.html#lazy-initial-state), but it's up to you.

Reply:
I'll leave as-is for now until the initial state gets expensive to calculate. At that point, this could be moved into a function and leverage lazy initial state.

Reply:
Upon further analysis, the position of the `ColorWell` thumb can be derived from props alone. I'm going to simplify this file and remove local component state.